nuget/spec/dependabot/nuget/file_updater_spec.rb

```diff
@@ -361,5 +361,53 @@
         end
       end
     end
+
+    context "with a dotnet-tools.json" do
+      let(:dependency_files) { [csproj_file, dotnet_tools_json] }
+      let(:dotnet_tools_json) do
+        Dependabot::DependencyFile.new(
+          content: fixture("dotnet_tools_jsons", "dotnet-tools.json"),
+          name: ".config/dotnet-tools.json"
+        )
+      end
+      let(:dependency) do
+        Dependabot::Dependency.new(
+          name: "dotnetsay",
```

Comment:
Since we're updating the last dependency in the file, I don't think this catches the conflict we discuss, we'd need to update the _first_ dependency in the file so that the conflict could happen.

Reply:
I changed the test files to make both tools have the same version the confirm in the spec that one one is updated and not both. Did I miss somthing?

Reply:
If you revert the changes in the regex, I think the current spec would still pass.

Because the match starts from `"#{dependency.name}"` all the way to `"#{dependency.previous_version}"` and the problem of accidentally matching versions of two dependencies because the version of the second dependency version is the same as the first one can only happen when updating the first dependency in the file (`botsnay`, not `dotnetsay`).

Let me know if I didn't explain this properly, regexps are hard 😅.

Reply:
Looks good to me now!